addons/hr_expense/models/hr_expense.py

```diff
@@ -435,7 +442,7 @@ def write(self, vals):
     @api.multi
     def unlink(self):
         for expense in self:
-            if expense.state == "post":
+            if expense.state in ['post', 'done']:
                 raise UserError(_("You cannot delete a posted expense."))
```

Comment:
I would have modified the message with something like `You cannot delete a posted or paid expense report.`